src/modules/acbs/acbs-party.service.test.ts

```diff
@@ -0,0 +1,145 @@
+import { HttpService } from '@nestjs/axios';
+import { PartyGenerator } from '@ukef-test/support/generator/party-generator';
+import { RandomValueGenerator } from '@ukef-test/support/generator/random-value-generator';
+import { AxiosError } from 'axios';
+import { when } from 'jest-when';
+import { of, throwError } from 'rxjs';
+
+import { AcbsPartyService } from './acbs-party.service';
+import { AcbsException } from './exception/acbs.exception';
+import { AcbsResourceNotFoundException } from './exception/acbs-resource-not-found.exception';
+
+describe('AcbsPartyService', () => {
+  const valueGenerator = new RandomValueGenerator();
+  const idToken = valueGenerator.string();
+  const baseUrl = valueGenerator.string();
+
+  let httpService: HttpService;
+  let service: AcbsPartyService;
+
+  beforeEach(() => {
+    httpService = new HttpService();
+    service = new AcbsPartyService({ baseUrl }, httpService);
+  });
+
+  describe('getPartyByIdentifier', () => {
+    const partyIdentifier = valueGenerator.stringOfNumericCharacters();
+
+    it('returns the party if ACBS responds with the party', async () => {
+      const { partiesInAcbs } = new PartyGenerator(valueGenerator).generate({ numberToGenerate: 1 });
+      const partyInAcbs = partiesInAcbs[0];
```

Comment:
Do we need to add any validation before accessing index `0` in case it is undefined?

Reply:
`PartyGenerator` is just a test helper that I wrote to create some values for tests to use so I don't think any validation is needed here